core/types/src/measurement.rs

```diff
@@ -4,30 +4,33 @@
 //!
 //! Different types are used for MrSigner and MrEnclave to prevent misuse.
 
-use crate::impl_newtype_for_bytestruct;
+use crate::{impl_display_and_debug_for_measurement, impl_newtype_for_bytestruct};
 use mc_sgx_core_sys_types::{sgx_measurement_t, SGX_HASH_SIZE};
+use serde::{Deserialize, Serialize};
 
 /// An opaque type for MRENCLAVE values
 ///
 /// A MRENCLAVE value is a chained cryptographic hash of the signed
 /// enclave binary (.so), and the results of the page initialization
 /// steps which created the enclave's pages.
-#[derive(Default, Debug, Clone, Eq, PartialEq)]
+#[derive(Default, Deserialize, Clone, Copy, Eq, Hash, PartialEq, Serialize)]
 #[repr(transparent)]
 pub struct MrEnclave(sgx_measurement_t);
 
 /// An opaque type for MRSIGNER values.
 ///
 /// A MRSIGNER value is a cryptographic hash of the public key an enclave
 /// was signed with.
-#[derive(Default, Debug, Clone, Eq, PartialEq)]
+#[derive(Default, Deserialize, Clone, Copy, Eq, Hash, PartialEq, Serialize)]
 #[repr(transparent)]
 pub struct MrSigner(sgx_measurement_t);
 
 impl_newtype_for_bytestruct! {
     MrEnclave, sgx_measurement_t, SGX_HASH_SIZE, m;
     MrSigner, sgx_measurement_t, SGX_HASH_SIZE, m;
 }
+impl_display_and_debug_for_measurement!(MrEnclave);
+impl_display_and_debug_for_measurement!(MrSigner);
```

Comment:
:upside_down_face: I think this is fine the way it is, but if the change is made above to add a `,` to the macro, this could be reduced to
```suggestion
impl_display_and_debug_for_measurement!(MrEnclave, MrSigner);
```

As the macro is currently written, one can do
```rust
impl_display_and_debug_for_measurement!(MrEnclave MrSigner);
```